### DISCUSSION TITLE
Danbooru Interrogate

### DISCUSSION THREAD

**Discussion by kklouzal** [Kilvoctu/aiyabot#196] [Ideas]
The ability to do a Danbooru Interrogate on init_image requests, and optionally feed the interrogation output immediately into a /draw command, appending the interrogation output to the end/beginning of the users custom prompt.

**Comment by Kilvoctu** [comment]
Can you provide a step-by-step workflow of this? 
I'm not sure I am completely understanding the process you're requesting.

**Reply by kklouzal** [reply]
Sure not a problem, it would go a little something like this:
/draw prompt:highres,girl walking on beach init_image:_image link_ interrogate:True
First, AiyaBot would first take and run a DeepBooru interrogate on the uploaded img2image.
Second, it would take the results from the interrogation (which might be something like "1girl, animal, beach, bird, black legwear, blue skirt, blue sky, boat, cloud, cloudy sky, condensation trail, day, evening, flock, gradient sky, horizon, island, kneehighs, lake, lighthouse, mountain, mountainous horizon, ocean, orange sky, outdoors, pier, pleated skirt, sand, scenery, school uniform, seagull, serafuku, shoes, shoes removed, shore, short hair, skirt, sky, socks, solo, sun, sunrise, sunset, twilight, water, watercraft, waves") and place this output text _before_ the original prompt the user provided to the /draw command.

So our new prompt would be: _highres, girl walking on beach, 1girl, animal, beach, bird, black legwear, blue skirt, blue sky, boat, cloud, cloudy sky, condensation trail, day, evening, flock, gradient sky, horizon, island, kneehighs, lake, lighthouse, mountain, mountainous horizon, ocean, orange sky, outdoors, pier, pleated skirt, sand, scenery, school uniform, seagull, serafuku, shoes, shoes removed, shore, short hair, skirt, sky, socks, solo, sun, sunrise, sunset, twilight, water, watercraft, waves_

Finally, it would run a normal init_image draw using the modified prompt from above.

Does this make sense?

**Reply by Kilvoctu** [reply]
I'm not sure I want to add more options to /draw; it's already very bloated. There is actually a project aiming to do the opposite https://github.com/users/Kilvoctu/projects/4  

Is there any other way to go about this without adding options to the command 🤔
